### ISSUE TITLE
Investigate problems created by Polymer 3.3.0

### ISSUE THREAD

**Issue by caalador** [vaadin/flow#5966]
Polymer version 3.3.0 atleast made grid not appear on the page of beverage buddy.
Polymer 3.3 might bring in other issues also and this should be investigated that what needs fixing.

**Comment by denis-anisimov** [comment]
`Categories` are shown correctly.
`Reviews`  list is not shown.

The reason is apparently is : categories view is made of built-in components (like `VerticalLyout` , standard HTML elements). and reviews list is a template.

Inspector shows that there is `reviews-list` element and it has `shadow-root` but it's empty.

**Comment by denis-anisimov** [comment]
That looks like some race condition issue.
If I add `reviews-list`  manually from the browser console the element is initialized.
I may assume that `reviews-list` is added too early when it's not yet defined.
Apparently it has not been an issue before Polymer 3.3 but now it is....

**Comment by denis-anisimov** [comment]
My last comment is wrong.
There is no race condition.
If I make a button when make  a `ReviewsList` element in Java side and add it to the page on click then the behavior is the same: the element is created but it's empty with empty shadow-root.
At the same time if I create an element via the browser console and append it then it works fine.

Looks like we have some issue in our client-side engine ....

**Comment by denis-anisimov** [comment]
The code works fine if I execute it using JS:
```
getUI().get().getPage().executeJs(
                "var a = document.createElement('reviews-list'); $0.appendChild(a);",
                getElement());
```

**Comment by denis-anisimov** [comment]
The identified issue is: `attachShadow`  in the `SimpleElementBindingStrategy` class.

It checks whether there is already a `shadowRoot` in the element and if there is no yet `shadowRoot` it creates it.

Apparently there is no yet `shadowRoot`  for the newly created Polymer element. The code creates it and Polymer doesn't reuse this shadow root after all.

I'm not sure what happened before Polymer `3.3.0` whether the `shadowRoot` has been made for the Polymer element right away (and has been reused in the result) or Polymer is reusing the shadow root.
It might be this can be just a bug in Polymer `3.3.0` as well.

`Beverage` can be fixed in a several ways:
- change the client-side `attachShadow` method: don't create shadow root at all.
- remove shadow root creation from the server side `IdMapper` class.

The first way is not an option in its direct way: we should be able to give a way to create a shadow root and add to it elements if needed . That what `attachShadow`  does: it creates a `shadowRoot` if needed and do binding inside it which means adding children and do binding for them.
So we still should support this code .
All those things are not needed in the beverage case.

As I understand there is no need in (implicit) accessing to shadow root for a polymer template at all.
This is what's done inside the `IdMapper` class.
It should be possible to avoid using  `getOrCreateShadowRoot` inside `IdMapper` implementation.
The method is used only once : in the  `getElementById` method.

Several notes about `getElementById` method:
- it's used inside  `mapComponentOrElement` only.
- it injects _server side_ element if there is a child identified by  `id` on the server side.
- I don't understand how it's applicable in our case: this code is called from the `PolymerTemplate` CTOR which means there can't be any server side element _YET_ added to the component (it's a CTOR: there is nothing yet available !).
- so I would say we improbable usecase and it should be just removed: we don't need `injectServerSideElement` method at all and its usage. Is it even tested ?
- Even if I missed something and we need  `injectServerSideElement` method we don't need to create a shadow root in the  `getElementById` impl: if there is no shadow root then apparently it has no any element.

As I understand the polymer element will create a shadow root of the element on itself.
So we should not create it at all.

So what we need to do:
- revisit the need of `injectServerSideElement` . If it's not tested/never used just remove it.
- depending on the previous : remove `getElementById` method at all or change its impl avoiding shadow root creation
- figure out what to do if there is a need to do something with shadow root (with a polymer template or not).

The last item can be a separate ticket.
beverage will be fixed by the first and second item.

In case someone really need to do with shadowRoot we need another investigation/fix for this: we need either postpone any action related to the shadow root when it appears (don't know how). Most likely we need to do this only in polymer element case. In other case  `shadowRoot` should behave in a usual way. Or may be as I said: this is just a bug in Polymer `3.3.0` : it doesn't reuse existing ` ` `shadowRoot` ...
